apps/dcos_net/src/dcos_net_mesos_listener.erl

```diff
@@ -986,14 +1027,108 @@ stream(Data, #state{pid=Pid, size=Size, buf=Buf}=State) ->
     Buf0 = <<Buf/binary, Data/binary>>,
     case byte_size(Buf0) of
         BufSize when BufSize >= Size ->
-            <<Head:Size/binary, Tail/binary>> = Buf0,
-            State0 = State#state{size=undefined, buf=Tail},
-            try jiffy:decode(Head, [return_maps]) of Obj ->
-                {next, Obj, State0}
-            catch error:Error ->
-                {error, Error}
-            end;
+            stream_decode(Buf0, Size, State);
         _BufSize ->
             httpc:stream_next(Pid),
             {next, State#state{buf=Buf0}}
     end.
+
+-spec(stream_decode(binary(), integer(), State) -> {error, term()} |
+     {next, State} | {next, jiffy:object(), State}
+         when State :: state()).
+stream_decode(Buf, Size, State) ->
+    <<Head:Size/binary, Tail/binary>> = Buf,
+    State0 = State#state{size=undefined, buf=Tail},
+    try
+        prometheus_counter:inc(mesos_listener, messages_total, [], 1)
+    catch error:_Error ->
+        ok
+    end,
+    try jiffy:decode(Head, [return_maps]) of Obj ->
+        {next, Obj, State0}
+    catch error:Error ->
+        {error, Error}
+    end.
+
+%%%===================================================================
+%%% Metrics functions
+%%%===================================================================
+
+-spec(init_metrics() -> ok).
+init_metrics() ->
+    init_metrics_pubsub(),
+    init_metrics_mesos_state(),
+    init_metrics_received().
+
+init_metrics_pubsub() ->
+    prometheus_summary:new([
+        {registry, mesos_listener},
+        {name, pubsub_duration_seconds},
+        {labels, []},
+        {help, "Time spent processing pubsub"}]).
+
+init_metrics_received() ->
+    prometheus_counter:new([
+        {registry, mesos_listener},
+        {name, bytes_total},
+        {labels, []},
+        {help, "Total number of bytes received"}]),
+    prometheus_counter:new([
+        {registry, mesos_listener},
+        {name, failures_total},
+        {labels, []},
+        {help, "Total number of failures"}]),
+    prometheus_boolean:new([
+        {registry, mesos_listener},
+        {name, listening_leader_boolean},
+        {labels, []},
+        {help, "Boolean indicating that listening is from a mesos leader. " ++
+            " 0 indicates standby master, 1 indicates leader."}]),
+    prometheus_counter:new([
+        {registry, mesos_listener},
+        {name, messages_total},
+        {labels, []},
+        {help, "Total number of messages"}]).
+
+init_metrics_mesos_state() ->
+    prometheus_gauge:new([
+        {registry, mesos_listener},
+        {name, agents_total},
+        {labels, []},
+        {help, "Total number of Agents"}]),
```

Comment:
nit: why `Agents`, `Frameworks`, `Tasks` start with capital letters?

Reply:
No good reason, fixed that.